File: android/foxui/src/main/java/io/gitee/stevenzack/foxui/FObject/Widget/FDialog.java
package io.gitee.stevenzack.foxui.FObject.Widget;

import android.app.AlertDialog;
import android.content.DialogInterface;

import fox.Fox;
import io.gitee.stevenzack.foxui.FObject.FObject;
import io.gitee.stevenzack.foxui.FoxActivity;

public class FDialog extends FObject {
    public AlertDialog.Builder v;
    public DialogInterface di;

    public FDialog(FoxActivity activity) {
        parentController = activity;
        v = new AlertDialog.Builder(parentController);
    }

    @Override
    public String getAttr(String attr) {
        return "";
    }

    @Override
    public String setAttr(String attr, String value, final String value2) {
        if (value == null)
            return "";
        switch (attr) {
            case "Title":
                v.setTitle(value);
                break;
            case "View":
                if (parentController.viewmap.containsKey(value)) {
                    v.setView(parentController.viewmap.get(value).view);
                }
                break;
            case "PositiveButton":
                try {
                    v.setPositiveButton(value, new DialogInterface.OnClickListener() {
                        @Override
                        public void onClick(DialogInterface dialog, int which) {
                            Fox.triggerFunction(parentController, value2, "", "", "");
                        }
                    });
                } catch (Exception e) {
                    e.printStackTrace();
                }
                break;
            case "NegativeButton":
                try {
                    v.setNegativeButton(value, new DialogInterface.OnClickListener() {
                        @Override
                        public void onClick(DialogInterface dialog, int which) {
                            Fox.triggerFunction(parentController, value2, "", "", "");
                        }
                    });
                } catch (Exception e) {
                    e.printStackTrace();
                }
                break;
            case "Show":
                di = v.show();
                break;
            case "Dismiss":
                if (di != null) {
                    di.dismiss();
                }
                break;
        }
        return "";
    }
}

File: fcore/widget/fdialog/fdialog.go
package fdialog

import "github.com/StevenZack/fox/fcore"

type FDialog struct {
	fcore.FBaseView
}

func New(a fcore.IActivity) *FDialog {
	v:=&FDialog{}
	v.A=a
	v.Vid=fcore.NewToken()
	v.VType= "Dialog"
	fcore.ViewMap.Set(v.Vid,v)
	v.A.NewObject(v.VType,v.Vid)
	return v
}


func (v *FDialog) Title(s string) *FDialog {
	v.A.SetAttr(v.Vid, "Title", s,"")
	return v
}
func (v *FDialog) Append(iv fcore.IBaseView) *FDialog {
	v.A.SetAttr(v.Vid, "View", iv.GetBaseView().Vid,"")
	return v
}
func (v *FDialog) PositiveButton(text string, onClick func(*FDialog)) *FDialog {
	fnId := v.Vid+"PositiveButtonClicked"
	fcore.EventMap.Set(fnId, func(activity fcore.IActivity, s string, s2 string, s3 string) string {
		onClick(v)
		return ""
	})
	v.A.SetAttr(v.Vid, "PositiveButton", text, fnId)
	return v
}

func (v *FDialog) NegativeButton(text string, onClick func(*FDialog)) *FDialog {
	fnId := v.Vid+"NegativeButtonClicked"
	fcore.EventMap.Set(fnId, func(activity fcore.IActivity, s string, s2 string, s3 string) string {
		onClick(v)
		return ""
	})
	v.A.SetAttr(v.Vid, "NegativeButton", text, fnId)
	return v
}
func (v *FDialog) Show() *FDialog {
	v.A.SetAttr(v.Vid, "Show", "","")
	return v
}
func (v *FDialog) Dismiss() *FDialog {
	v.A.SetAttr(v.Vid, "Dismiss", "","")
	return v
}
